Xamarin.Forms.Core/NavigationPage.cs

```diff
@@ -251,6 +251,17 @@ protected override bool OnBackButtonPressed()
 
 			var page = (Page)InternalChildren.Last();
 
+			return await RemoveAsyncInner(page, animated, fast);
+		}
+
+		[EditorBrowsable(EditorBrowsableState.Never)]
+		public async Task<Page> RemoveAsyncInner(Page page, bool animated, bool fast)
```

Comment:
make it internal

Reply:
Can't make internal, but maybe put this on `INavigationPageController` in `Internals` namespace and implement explicitly here?